staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -204,12 +205,15 @@ func (s *sharedIndexInformer) Run(stopCh <-chan struct{}) {
 
 		s.controller = New(cfg)
 		s.controller.(*controller).clock = s.clock
+		s.stopCh = stopCh
 		s.started = true
 	}()
 
-	s.stopCh = stopCh
```

Comment:
This looks like a bug to me - this field is written here while not holding the startedLock lock and read while holding that lock in `AddEventHandlerWithResyncPeriod`. Looks like a potential data race. So I moved it into the locked region.